modules/common/util/ini.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2025 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// IniOption -
+type IniOption struct {
+	Section string
+	Key     string
+	Value   string
+}
+
+// Define static errors
+var (
+	ErrKeyAlreadyExists     = errors.New("key already exists in section")
+	ErrCouldNotPatchSection = errors.New("could not patch target section")
+)
+
+// repr - print key: value in .ini format
+func (i *IniOption) repr() string {
+	return fmt.Sprintf("%s = %s", i.Key, i.Value)
+}
+
+// ExtendCustomServiceConfig - customServiceConfig is tokenized and parsed in a
+// loop where we keep track of two indexes:
+//   - index: keep track of the current extracted token
+//   - sectionIndex: when we detect a [<section>] within a token, we save the index
+//     and we update it with the next section when is detected. This way we can
+//     make sure to evaluate only the keys of the target section
+//
+// when an invalid case is detected, we return the customServiceConfig string
+// unchanged, otherwise the new key=value is appended as per the IniOption struct
+func ExtendCustomServiceConfig(
+	iniString string,
+	customServiceConfigExtend IniOption,
+) (string, error) {
+	// customServiceConfig is empty
+	if len(iniString) == 0 {
+		return iniString, nil
+	}
+	// Position where insert new option (-1 = target section not found)
+	index := -1
+	// Current section header position (-1 = no section found)
+	sectionIndex := -1
+	svcConfigLines := strings.Split(iniString, "\n")
+	sectionName := ""
+	for idx, rawLine := range svcConfigLines {
+		line := strings.TrimSpace(rawLine)
+		token := strings.TrimSpace(strings.SplitN(line, "=", 2)[0])
+
+		if token == "" || strings.HasPrefix(token, "#") {
+			// Skip blank lines and comments
+			continue
+		}
+		if strings.HasPrefix(token, "[") && strings.HasSuffix(token, "]") {
+			// Note the section name before looking for a backend_name
+			sectionName = strings.Trim(token, "[]")
+			sectionIndex = idx
+			// increment the index (as an offset) only when a section is found
+			if sectionName == customServiceConfigExtend.Section {
+				index = idx + 1
+			}
+		}
+		// Check if key already exists in target section
+		if token == customServiceConfigExtend.Key && sectionIndex > -1 &&
+			sectionName == customServiceConfigExtend.Section {
+			errMsg := fmt.Errorf("%w: key %s in section %s", ErrKeyAlreadyExists, token, sectionName)
```

Comment:
the openstack config is not strictly speaking ini config where you always have only once a parameter (key) in a section. there are services where it is valid to have multiple keys with the same name. e.g. in nova for pci passthrough, like the pci_passthrough_alias parameter.